File: backend/buylogic/src/main/java/com/buylogic/model/StockMovement.java
package com.buylogic.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "stock_movement")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class StockMovement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idStockMovement;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_product", nullable = false)
    private Product product;

    @Column(name = "movement_type", nullable = false, length = 30)
    private String movementType;

    @Column(nullable = false, precision = 14, scale = 3)
    private BigDecimal quantity;

    @Column(name = "movement_date", nullable = false)
    private LocalDateTime movementDate;

    @Column(length = 100)
    private String reference;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        if (movementDate == null) {
            movementDate = LocalDateTime.now();
        }

        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/stockMovement.ts
export type StockMovementType =
  | "PURCHASE"
  | "SALE"
  | "RETURN"
  | "LOSS"
  | "ADJUSTMENT"
  | "TRANSFER";

export interface StockMovement {
  idStockMovement: number;
  idProduct: number;
  movementType: StockMovementType;
  quantity: number;
  movementDate: string;
  reference: string;
  createdAt?: string;
}

export interface StockMovementCreate {
  idProduct: number;
  movementType: StockMovementType;
  quantity: number;
  movementDate?: string;
  reference?: string;
}

export interface StockAdjustment { 
  targetStock: number; 
  reason: string; 
}